packages/console/src/pages/ApplicationDetails/index.tsx

```diff
@@ -1,46 +1,149 @@
 import { Application } from '@logto/schemas';
-import React from 'react';
+import React, { useEffect } from 'react';
+import { useForm } from 'react-hook-form';
 import { useTranslation } from 'react-i18next';
-import { useParams } from 'react-router-dom';
+import { useLocation, useParams } from 'react-router-dom';
 import useSWR from 'swr';
 
 import BackLink from '@/components/BackLink';
 import Button from '@/components/Button';
 import Card from '@/components/Card';
 import CopyToClipboard from '@/components/CopyToClipboard';
+import FormField from '@/components/FormField';
 import ImagePlaceholder from '@/components/ImagePlaceholder';
+import TabNav, { TabNavLink } from '@/components/TabNav';
+import TextInput from '@/components/TextInput';
 import { RequestError } from '@/swr';
 import { applicationTypeI18nKey } from '@/types/applications';
 
 import * as styles from './index.module.scss';
 
+type OidcConfig = {
+  authorization_endpoint: string;
+  userinfo_endpoint: string;
+  token_endpoint: string;
+};
```

Comment:
Would it be better to define this type in `schemas` package?

Reply:
extract to the other place other than `schemas` package?

Reply:
better re-use `OidcConfigSnakeCaseResponse` in `js/js` package?

Reply:
This is not a schema.
It returns from the oidc provider and currently, we can not share the codes between the `core` and the `js` package.
@IceHe

Reply:
OK, I see. We can consider exporting this type from js sdk, as we are going to integrate React sdk in #373. So technically we can export this type all the way from core js sdk to React sdk.
I can add a task to do that tomorrow.

Reply:
Added: LOG-1876

Reply:
Is just a form type, I think is OK to define a type here.